src/FixtureBuilder/ExpressionLanguage/Parser/TokenParser/Chainable/FunctionTokenParser.php

```diff
@@ -129,13 +128,49 @@ function (array $matches) use ($argumentEscaper): string {
         );
 
         $arguments = [];
+
         preg_match_all('/\[[^[]+\]|[^,\s]+/', $escapedString, $argumentsList);
-        foreach ($argumentsList[0] as $index => $argument) {
-            $argument = $parser->parse($argument);
 
-            $arguments[$index] = $argument;
+        foreach ($argumentsList[0] as $index => $argument) {
+            $arguments[$index] = $this->parseArgument($parser, $argument);
         }
 
         return $arguments;
     }
+
+    private function parseArgument(ParserInterface $parser, string $value)
+    {
+        switch (true) {
+            case $value === 'true':
+                return true;
+            case $value === 'false':
+                return false;
+            case $value === 'null':
+                return null;
+            case ctype_digit($value):
+                $castedValue = (int) $value;
+
+                if (0 === $value[0]) {
+                    return octdec($value);
+                } elseif ($value === (string) $castedValue) {
```

Comment:
no need for an else if your return before

Reply:
There are three different cases to cover: I have to either split them into `if/elseif/else` statements or into multiple `if` statements 🤔 I'm not sure I get what I should change

Reply:
Instead of:

```php
if (x) {
  // ...
  return ...;
} elseif (y) {
  // ...
  return ...;
} else {
  // ...
  return ...;
}

```

A strict equivalent is:

```php
if (x) {
  // ...
  return ...;
}

if (y) {
  // ...
  return ...;
}
  
// ...
return ...;

```